### PR TITLE
check for install using virtualenv

### PR DESCRIPTION
**Pull request by tejalwakchoure**
## What? Why?
installation using virtualenv. (Add Reference to the issue if any)

## Checks
- [ ] Single commit and [No merge commits](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Make sure you have added required documentation
- [ ] If you have refactored a code, make sure that you have compared the outputs pre and post refactoring. Add both pre and post refaoring screenshots below if the output is visual

## Notify reviewers
@achyudhk @prasadtalasila

### CODE REVIEW

File: etc/install_package_dependencies.sh [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
+sudo apt-get install -y gtk 3
 sudo apt-get install -y libboost-all-dev
 sudo apt-get install -y build-essential
 sudo apt-get install -y libcairo2-dev
-sudo apt-get install -y python3-pip
-sudo apt-get install -y python3-dev
-sudo apt-get install -y python3-matplotlib
 sudo apt-get install -y gfortran libopenblas-dev liblapack-dev
 sudo apt-get install -y libcgal-dev
-sudo apt-get install -y python3-numpy
```
Comment (prasadtalasila) [inline etc/install_package_dependencies.sh:22]: I am presuming that these modules have been moved to requirements.txt. Is it correct?

File: etc/install_package_dependencies.sh [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
-sudo apt-get install -y python2.7-config
-sudo apt-get install -y python3-cairo
-sudo apt-get install -y python3-scipy
 sudo apt-get install -y graphviz graphviz-dev libgraphviz-dev pkg-config
-sudo apt-get install -y python3-pygraphviz
```
Comment (prasadtalasila) [inline etc/install_package_dependencies.sh:27]: Is this a pure python library? If so, has this been moved to requirements.txt?

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@tejalwakchoure and @AakankshaSanctis Has the installation been tried on a clean computer and found to be working?

File: etc/install_package_dependencies.sh (hunk 1 above)
Comment (tejalwakchoure) [reply etc/install_package_dependencies.sh:22]: The modules that have been removed were already present in requirements.txt

File: etc/install_package_dependencies.sh (hunk 2 above)
Comment (tejalwakchoure) [reply etc/install_package_dependencies.sh:27]: Yes, it's present in requirements.txt

File: etc/install_package_dependencies.sh [hunk 3]
```diff
@@ -1,30 +1,28 @@
...
 
 # Install Python-PIP
 sudo apt-get update
-sudo apt-get install -y python-pip
+sudo apt-get install -y python3-pip
+
+# Install virtualenv, create a virtual environment and activate it
+sudo -H pip3 install virtualenv
+virtualenv env
+source env/bin/activate
```
Comment (hrishikesh-dahiya) [inline etc/install_package_dependencies.sh]: can the virtualenv be installed in .env instead of env?

Comment (prasadtalasila) [reply etc/install_package_dependencies.sh]: The idea makes sense. Also the `.env` directory needs to be added to `.gitignore`.

File: etc/install_package_dependencies.sh [hunk 4]
```diff
@@ -1,30 +1,28 @@
...
 
 
 # Install Graph-Tool
 sudo apt-get install -y expat
 sudo apt-get install -y libsparsehash-dev
-sudo apt-get install -y gtk+3
+sudo apt-get install -y gtk 3
```
Comment (prasadtalasila) [inline etc/install_package_dependencies.sh]: This change in package name from "gtk+3" to "gtk 3" leads to installation failure. If we need gtk-3 to be installed, then "gtk+3" or "libgtk-3-dev" seems to be the packages.

Comment (prasadtalasila) [inline etc/install_package_dependencies.sh]: This change in package name from "gtk+3" to "gtk 3" leads to installation failure. If we need gtk-3 to be installed, then "gtk+3" or "libgtk-3-dev" seems to be the packages.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]

File: etc/install_package_dependencies.sh (hunk 3 above)
Comment (AakankshaSanctis) [reply etc/install_package_dependencies.sh]: .env directory already present in .gitignore

**Comment by tejalwakchoure** [comment]
@prasadtalasila venv is apparently preferred over virtualenv, so I have changed the installation to venv. Have also added python 3.5 installation commands, as discussed.

**Comment by prasadtalasila** [comment]
PR #136 improves and includes all the work in this PR. Hence closing this PR.

**Pull request closed without merging**